Chutzpah/TestRunner.cs

```diff
@@ -305,10 +305,14 @@ private IChutzpahWebServerHost SetupWebServerHost(ConcurrentBag<TestContext> tes
             if (contextWithChosenServerConfiguration != null)
             {
                 var webServerConfiguration = contextWithChosenServerConfiguration.TestFileSettings.Server;
-                webServerHost = webServerFactory.CreateServer(webServerConfiguration, ActiveWebServerHost);
 
-                // Stash host object on context for use in url generation
-                contextUsingWebServer.ForEach(x => x.WebServerHost = webServerHost);
+                if (webServerConfiguration.Enabled.GetValueOrDefault())
+                {
+                    webServerHost = webServerFactory.CreateServer(webServerConfiguration, ActiveWebServerHost);
+
+                    // Stash host object on context for use in url generation
+                    contextUsingWebServer.ForEach(x => x.WebServerHost = webServerHost);

```

Comment:
I think this is still wrong though since before this would just be the set of contexts that are set up for web server mode, but now this is all webservers that have the setting true OR false.

Reply:
if (webServerConfiguration.Enabled.GetValueOrDefault())
This will enter the block only for enabled, right?

Reply:
Yes that means that there is at least ONE context that is enabled for server, but then here you are setting that on all contexts that have a server block, even if they are disabled?

Reply:
Also, contextUsingWebServer is a wrong name to that variable with this change since it doesn't mean they are using the web server

Reply:
Good point. Do you have UT to cover all these already? I would like to run them all since I am not familiar with the whole code base.
I ran "build" on command line and looks like it did pass all tests.